libbeat/monitoring/inputmon/input.go

```diff
@@ -19,23 +19,115 @@ package inputmon
 
 import (
 	"encoding/json"
+	"errors"
 	"strings"
+	"sync"
 
 	"github.com/gofrs/uuid/v5"
 
 	"github.com/elastic/elastic-agent-libs/logp"
 	"github.com/elastic/elastic-agent-libs/monitoring"
 )
 
-// NewInputRegistry returns a new monitoring.Registry for metrics related to
-// an input instance. The returned registry will be initialized with a static
-// string values for the input and id. When the input stops it should invoke
-// the returned cancel function to unregister the metrics. For testing purposes
-// an optional monitoring.Registry may be provided as an alternative to using
-// the global 'dataset' monitoring namespace. The inputType and id must be
-// non-empty for the metrics to be published to the global 'dataset' monitoring
-// namespace.
-func NewInputRegistry(inputType, id string, optionalParent *monitoring.Registry) (reg *monitoring.Registry, cancel func()) {
+type inputRegistry struct {
+	mu         sync.RWMutex
+	registries map[string]*monitoring.Registry
+}
+
+var registeredInputs = inputRegistry{
```

Comment:
I hate to add a new global table just as we're transitioning to a single-process design that requires us to stop using global tables. Can we create this somewhere more contextual and maybe attach it to the handler? The handler itself is still global for now but it'll be easier to clean up later if the new features are at least propagated with local parameters.

Reply:
I thought about it as well. But I rather do it on another PR instead of adding more changes to this PR

Reply:
> I thought about it as well. But I rather do it on another PR instead of adding more changes to this PR

I'd rather this PR not add a new global lookup table in the first place. Unless there's some concrete reason why this would be a large or infeasible change (and it doesn't look that way to me), please make it. Cleaning up the design of a new feature is what code review is for, and followup PRs aren't a good rationale for merging broken code. The only reason this didn't come up in an earlier pass is that it didn't make sense to focus on more localized issues without a working top-level design, and just because the top-level design took a while doesn't mean we should skip the rest of the review.